policyengine_us/variables/gov/states/de/tax/income/deductions/de_itemized_deductions.py

```diff
@@ -0,0 +1,44 @@
+from policyengine_us.model_api import *
+
+
+class de_itemized_deductions(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Delaware itemized deductions"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=7"
+        "https://revenuefiles.delaware.gov/2022/PIT-RES_TY22_2022-02_Instructions.pdf#page=11"
+        "https://casetext.com/statute/delaware-code/title-30-state-taxes/part-ii-income-inheritance-and-estate-taxes/chapter-11-personal-income-tax/subchapter-ii-resident-individuals/section-1109-itemized-deductions-for-application-of-this-section-see-66-del-laws-c-86-section-8"
+    )
+    defined_for = StateCode.DE
+
+    def formula(tax_unit, period, parameters):
+        # 2021 Form M1 instructions say:
+        #   You may claim the Delaware standard deduction or itemize
+        #   your deductions on your Delaware return. You will generally
+        #   pay less Delaware income tax if you take the larger of your
+        #   itemized or standard deduction.
+        # ... calculate pre-limitation itemized deductions
+        itm_deds_less_salt = tax_unit("itemized_deductions_less_salt", period)
+        # ... calculate itemized deductions offset
+        p = parameters(period).gov.states.mn.tax.income.deductions
+        exempt_deds = add(
+            tax_unit,
+            period,
+            [
+                "medical_expense_deduction",
+                "casualty_loss_deduction",
+                "charitable_deduction",
+                "salt_deduction",
+            ],
+        )
+        net_deds = max_(0, itm_deds_less_salt - exempt_deds)
+        net_deds_offset = p.deduction_fraction * net_deds
+        agi = tax_unit("adjusted_gross_income", period)
```

Comment:
we have to prorate the itemized deductions for separate filers - deduction is % of each spouses % of total income
![image](https://github.com/PolicyEngine/policyengine-us/assets/110687043/aa730bb2-8f42-4f2f-a781-55a8232c14fb)